### DISCUSSION TITLE
Detect open TCP ports

### DISCUSSION THREAD

**Discussion by JaneX8** [projectdiscovery/nuclei#2940] [Q&A] [answered]
Is it possible to detect open TCP ports? I understand I can match a specific reply but what is the reply is unknown and I simply want to determine: **any** or **no** reply.

**Comment by Ice3man543** [comment]
@JaneX8 there is a potential to add a fast SYN based port scanner to nuclei as part of an optional implementation that allows in-built port scanning. There has been discussion on it in the past but no concrete steps have been taken yet. It's a very interesting idea though. cc @ehsandeep

**Comment by JaneX8** [comment]
This could make existing templates better and open up the possibilities for many more detections.

**Comment by olearycrew** [comment, marked as answer]
For finding open TCP ports, we recommend using our tool Naabu: https://nux.gg/naabu

**Comment by JaneX8** [comment]
@olearycrew That's great, but it's more about using open port detection in other Nuclei scripts.
